### DISCUSSION TITLE
Multiple windows?

### DISCUSSION THREAD

**Discussion by posguy99** [joelekstrom/fastmate#65] [Ideas]
Something running Fastmail in a Fluid SSB allows you to do is have multiple windows.  You can hit the drop down while looking at your mail, and instead of clicking the Calendar icon to switch views, you can right-click the Calendar icon and pop it out into another window.

I can't make that work in Fastmate, though.  I end up with it opening a tab in Safari and being asked for a Fastmail login.

Failing that, how about being able to specify the view the application opened into?  I'm not sure how you'd control that, though, since it'd be problematical to somehow have copy (A) launch one way, and copy (B) launch another way.  If it weren't macOS, I'd say a command-line option to the binary, but bundles and bundle identifiers get in the way.

Just an idea.

**Comment by joelekstrom** [comment]
When I designed Fastmate I wanted it to be a single page experience (built it mostly for myself), so it's explicitly built like that. However, I've seen some people want multiple windows. For example, @mdbraber built his own fork that allows opening the compose view in a different window, which I actually think sounds really nice. (#14)

I'll experiment a bit and see if it can support multiple windows in a nice way!

**Comment by iBook140** [comment]
Hello guys,

sorry for replaying to an old thread, but I’m searching for exactly this possibility to open new messages in new window (like Outlook or most email clients do).


Can you please tell me where to finde this fork from @mdbraber?

Kind regards
René

**Reply by mdbraber** [reply]
The branch is here: https://github.com/mdbraber/fastmate/tree/add_compose_window - I don't use it myself and haven't looked at it in 2 years so it might be totally unusable but possibly a starting point if you know how to code

**Reply by iBook140** [reply]
Hello @mdbraber 

thank you very much, but unfortunately I don’t know how to code or compile an MacOS app from this ☹️ 
Is there any ready repository (which I just can use as .app) I can try?

Kind regards
René

**Reply by mdbraber** [reply]
No unfortunately there isn't - you need to load the code in Xcode, adapt a few settings and try to compile it. Unfortunately I don't have time to explain in more detail. But even then: the code is uncompleted and too old so I wouldn't recommend using it.

**Reply by iBook140** [reply]
I managed it to build a version with Xcode and new window for composing new email seems to work. Also my test email etc. worked just fine 😊 

May I ask why you didn’t develop this further? For example closing window if email is send would be nice to have.

PS Due to missing window feature in Fastmail I switched to MS 365/Outlook 🫣

Kind regards
René

**Reply by mdbraber** [reply]
Great to hear; it's just a matter of time and priorities. If someone wants to take it forward at this moment I think it's still a nice idea
